roles/cifmw_cephadm/tasks/pools.yml

```diff
@@ -47,19 +47,21 @@
 
 - name: Configure the RBD trash purge scheduler
   when:
-    - cifmw_enabled_services | default([]) | intersect(['cinder_volume'])
     - cifmw_cephadm_enable_trash_scheduler | default(false)
+    - cifmw_cephadm_pools is defined
+    - cifmw_cephadm_pools | length > 0
   block:
     - name: Get the RBD ceph_cli
       ansible.builtin.include_tasks: ceph_cli.yml
       vars:
         ceph_command: rbd
 
     - name: Set trash interval
+      when: item.application == 'rbd'
       ansible.builtin.command:
-        cmd: |
-          {{ cifmw_cephadm_ceph_cli }} trash purge schedule add \
-          {{ cifmw_cephadm_rbd_trash_interval | default(15) }} --pool {{ item }}
+        cmd: >-
+          {{ cifmw_cephadm_ceph_cli }} trash purge schedule add
+          {{ cifmw_cephadm_rbd_trash_interval | default(15) }} --pool {{ item.name }}
       changed_when: false
       become: true
-      loop: "{{ [ cinder_pool.name | default('volumes') ] + cinder_pool.cinder_extra_pools | default([]) }}"
+      loop: "{{ cifmw_cephadm_pools | default([]) }}"
```

Comment:
Is the goal to set the rbd trash purge scheduler to all the ceph pools w/ an rbd application set? Or the goal was to enabled it only for cinder pools?

Reply:
you are right, i think goal here is set it only for cinder pools

in tripleo, cinder_pool was defined in `ceph_pools ` like this
ceph_pools:
  cinder_pool:
    name: 'volumes'
    enabled: true
    cinder_extra_pools: [altrbd, pool2, pool3]

in cifmw , we have

cifmw_cephadm_pools:
- name: volumes
  pg_autoscale_mode: True
  target_size_ratio: 0.3
  application: rbd

@fultonj 
so we have only one pool `volumes` for cinder, should we set the rbd trash purge scheduler only for that pool and remove the loop ?

Reply:
I would only set for volumes **if** is part of the `cifmw_cephadm_pools` (this should be the when condition). Right now this is wrong because you're setting the trash scheduler for all pools, and I don't think this was the original goal.

Reply:
+1 to what fmount said.

Reply:
@fmount agree with you, code is now updated and it sets trash purge scheduler only to volumes pool.